yt_dlp/extractor/vanillo.py

```diff
@@ -0,0 +1,386 @@
+import datetime
+import json
+import re
+
+from .common import InfoExtractor
+from ..utils import ExtractorError, parse_iso8601
+
+# NOTE: Private videos can be downloaded by adding --add-header "authorization: Bearer abcxyz",
+# but won't work with --cookies-from-browser and --cookies file.txt
+
+
+class VanilloIE(InfoExtractor):
+    _access_token = None
+
+    def _real_initialize(self):
+        """
+        Called before extracting each URL. This checks the cookie jar for an access token
+        """
+        # 1) Check if we already have an access token in our cookies
+        if not self._access_token:
+            self._access_token = self._get_token_from_cookies()
+
+        # If we still have no token, requests remain unauthorized (public only).
+
+    def _get_token_from_cookies(self):
+        """
+        Attempt to read an access_token from cookies (if any).
+        """
+        cookie_names = ('access_token', 'ACCESS_TOKEN')
+        # We rename 'path' to '_path' since it's unused in the loop body
+        for domain, path_dict in self._downloader.cookiejar._cookies.items():
+            if 'vanillo.tv' in domain:
+                for _path, cookie_dict in path_dict.items():
+                    for name, cookie in cookie_dict.items():
+                        if name in cookie_names:
+                            # self.to_screen(f'Found vanillo.tv access_token in cookies: {cookie.value}')
+                            self.to_screen('Found vanillo.tv access_token in cookies')
+                            return cookie.value
+        return None
```

Comment:
This can be simplified. (Also FYI: `_real_initialize` is only called when the extractor is initialized, not before each URL. If the user has passed multiple URLs, a batch file, or a playlist URL, the same extractor instance will be used for each URL)
```suggestion
    def _real_initialize(self):
        if self._access_token:
            return
        self._access_token = traverse_obj(
            self._get_cookies('https://vanillo.tv/'), ('access_token', 'value'), casesense=False)
        if self._access_token:
            self.write_debug('Found vanillo.tv access token in cookies')
```